### PR TITLE
1st task of testing

### PR DESCRIPTION
**Pull request by IvayloTsvetkov**
Tests for Sumator's sum method when legal and illegal arguments are passed.

### CODE REVIEW

File: testing_1/test/com/company/SumatorTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    s = new Sumator();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void sumWhenIllegalArgumentIsPassed() {
+    s.sum("64", "a");
+  }
+
+  @Test
+  public void sumOfLegalArguments() {
```
Comment (mgenov) [inline testing_1/test/com/company/SumatorTest.java]: Legal/illegal valid/invalid are not good keywords for tests. 

There is another technique which you can use here. If you name your test like: 

```java
class SumNumbersAsStringsTest {
   
   @Test
   public void happyPath() {
   
    }
}
```
e.g the happy path for summing of numbers as strings.

File: testing_1/test/com/company/SumatorTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.company;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+public class SumatorTest {
+  Sumator s;
```
Comment (mgenov) [inline testing_1/test/com/company/SumatorTest.java]: private Sumator sumator = new Sumator(); 

The before statement is not required for such simple cases .

File: testing_1/test/com/company/SumatorTest.java (hunk 1 above)
Comment (IvayloTsvetkov) [inline testing_1/test/com/company/SumatorTest.java]: What about the test with the illegal argument ? Should i leave it like that or change it to something else like unhappyPath ?

Comment (mgenov) [inline testing_1/test/com/company/SumatorTest.java]: Happy and Unhappy sounds silly. 

You create separate test which is concrete about this scenario. Keep in mind that you don't have only 2 tests per test class. 

Here is a sample test from oauth2 server library that could give you some light:
```java
/**
 * @author Miroslav Genov (miroslav.genov@clouway.com)
 */
public class DecodeClientCredentialsTest {

  @Rule
  public JUnitRuleMockery context = new JUnitRuleMockery();

  private ClientRequest clientRequest = context.mock(ClientRequest.class);

  private ClientAuthenticationCredentialsRequest handler = new ClientAuthenticationCredentialsRequest(clientRequest);

  @Test
  public void happyPath() {
    final DateTime anyInstantTime = new DateTime();
    final Request clientAuthRequest = clientAuthRequest(
            "Basic ZmU3MjcyMmE0MGRlODQ2ZTAzODY1Y2IzYjU4MmFlZDU3ODQxYWM3MTo4NTc2MTNkYjdiMTgyMzJjNzJhNTA5M2FkMTlkYmM2ZGY3NGExMzll"
    );

    context.checking(new Expectations() {{
      oneOf(clientRequest).handleAsOf(
              clientAuthRequest, new ClientCredentials(
                      "fe72722a40de846e03865cb3b582aed57841ac71",
                      "857613db7b18232c72a5093ad19dbc6df74a139e"
              ), anyInstantTime);
    }});


    handler.handleAsOf(clientAuthRequest, anyInstantTime);
  }

  @Test
  public void headerIsNotForBasicAuthorization() {
    final Request clientAuthRequest = clientAuthRequest(
            "::some header value::"
    );
    Response response = handler.handleAsOf(clientAuthRequest, anyInstantTime());
    assertThat(response.status().code, is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
  }

  @Test
  public void basicHeaderIsNotEncodedWithBase64() {
    final Request clientAuthRequest = clientAuthRequest(
            "Basic z"
    );
    Response response = handler.handleAsOf(clientAuthRequest, anyInstantTime());
    assertThat(response.status().code, is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
  }

  @Test
  public void clientIdAndClientSecretAreNotCorrectlySeparated() {
    final Request clientAuthRequest = clientAuthRequest(
            "Basic dGVzdGlkdGVzdHNlY3JldA==" // is encoding of: testidtestsecret
    );
    Response response = handler.handleAsOf(clientAuthRequest, anyInstantTime());
    assertThat(response.status().code, is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
  }

  @Test
  public void authenticatePublicClient() throws Exception {
    final Request clientAuthRequest = aNewRequest().param("client_id", "::client id::").build();
    final DateTime anyInstantTime = new DateTime();

    context.checking(new Expectations() {{
      oneOf(clientRequest).handleAsOf(
              clientAuthRequest, new ClientCredentials(
                      "::client id::",
                      ""
              ), anyInstantTime);
    }});

    handler.handleAsOf(clientAuthRequest, anyInstantTime);
  }

  private DateTime anyInstantTime() {
    return new DateTime();
  }

  private Request clientAuthRequest(String authorizationValue) {
    return new ByteRequest(
            ImmutableMap.<String, String>of(),
            ImmutableMap.of("Authorization", authorizationValue)
    );
  }

}
```

**Comment by IvayloTsvetkov** [comment]
Fixed.

File: testing_1/test/com/company/SumatorTest.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class SumatorTest {
+  private Sumator s = new Sumator();
+
+  @Test
+  public void happyPath() {
+    assertThat(s.sum("5", "6"), equalTo("11"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void sumWhenCharacterIsPassed() {
```
Comment (mgenov) [inline testing_1/test/com/company/SumatorTest.java]: sumWhenLetterIsPassedIn

**Comment by IvayloTsvetkov** [comment]
Fixed.

File: testing_1/test/com/company/SumatorTest.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.company;
+
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+public class SumatorTest {
```
Comment (mgenov) [inline testing_1/test/com/company/SumatorTest.java]: If you have test named happyPath, then your test class cannot be named SumatorTest.

**Comment by IvayloTsvetkov** [comment]
Fixed and added test when Null is passed.

**Pull request merged**